org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/compiler/OnTheFlyJavaCompiler2.java

```diff
@@ -0,0 +1,187 @@
+ * 
+ * @since 2.9
+ * @deprecated Use org.eclipse.xtext.xbase.testing.OnTheFlyJavaCompiler2 instead
+ */
+@Deprecated
+@Singleton
+public class OnTheFlyJavaCompiler2 {
+	private InMemoryJavaCompiler inMemoryCompiler;
+
+	/**
+	 * Creates a new OnTheFlyCompiler that accepts Java6 compliant code.
+	 */
+	@Inject
+	public OnTheFlyJavaCompiler2(ClassLoader scope) {
+		this(scope, JavaVersion.JAVA6);
+	}
+
+	public OnTheFlyJavaCompiler2(ClassLoader scope, JavaVersion version) {
+		this.inMemoryCompiler = new InMemoryJavaCompiler(scope, version);
+	}
+
+	public Class<?> compileToClass( String classname,  String code) {
+		InMemoryJavaCompiler.Result result = inMemoryCompiler.compile(new JavaSource(toJavaFile(classname), code));
+		try {
+			if (IterableExtensions.exists(result.getCompilationProblems(), CategorizedProblem::isError)) {
+				StringConcatenation richString = new StringConcatenation();
+				richString.append("Java code compiled with errors:");
+				richString.newLine();
+				richString.append(Joiner.on("\n")
+						.join(IterableExtensions.filter(result.getCompilationProblems(), CategorizedProblem::isError)));
+				richString.newLineIfNotEmpty();
+				richString.newLine();
+				richString.append("Code was:");
+				richString.newLine();
+				richString.append(code);
+				richString.newLineIfNotEmpty();
+				throw new IllegalArgumentException(richString.toString());
+			}
+			return result.getClassLoader().loadClass(classname);
+		} catch (Throwable t) {
+			if (t instanceof ClassNotFoundException) {
+				ClassNotFoundException e = (ClassNotFoundException) t;
+				StringConcatenation richString = new StringConcatenation();
+				richString.append("Couldn\'t load \'");
+				richString.append(classname);
+				richString.append("\' ");
+				richString.newLineIfNotEmpty();
+				richString.append("source :");
+				richString.newLine();
+				richString.append("\t");
+				richString.append(code, "\t");
+				richString.newLineIfNotEmpty();
+				richString.newLine();
+				richString.append("PROBLEMS : ");
+				richString.newLine();
+				richString.append("\t");
+				richString.append(Joiner.on("\n").join(result.getCompilationProblems()), "\t");
+				richString.newLineIfNotEmpty();
+				throw new IllegalStateException(richString.toString(), e);
+			} else {
+				throw Exceptions.sneakyThrow(t);
+			}
+		}
+	}
+
+	private String toJavaFile(String string) {
+		return string.replace(".", "/") + ".java";
+	}
+
+	public Map<String, Class<?>> compileToClasses(Map<String, String> sources) {
+		InMemoryJavaCompiler.Result result = inMemoryCompiler
+				.compile(((JavaSource[]) Conversions.unwrapArray(
+						Iterables.transform(sources.entrySet(),
+								(Map.Entry<String, String> it) -> new JavaSource(toJavaFile(it.getKey()), it.getValue())),
+						JavaSource.class)));
+		try {
+			if (IterableExtensions.exists(result.getCompilationProblems(), CategorizedProblem::isError)) {
+				StringConcatenation richString = new StringConcatenation();
```

Comment:
Just an error text, use ordinary string concatenation!?

Reply:
did not want to change any semantics on a deprecated class